src/pages/Posts.js

```diff
@@ -13,25 +13,43 @@ const Posts = () => {
       .then((data) => setPosts(data));
   }, []);
 
-  const selectedPost = posts.filter((item) => {
-    if (loggedInUser.id === item.userId) {
-      return item;
+
+  const allUsersAndPosts= [
+    {
+      name: "",
+      post: "",
+      title: "",
+    },
+  ];
+  for (let i = 0; i < allUser.length; i++) {
+    for (let k = 0; k < posts.length; k++) {
+      if (allUser[i].id === posts[k].userId) {
+        allUsersAndPosts.push({
+          name: allUser[i].name,
+          post: posts[k].body,
+          title: posts[k].title,
+        });
+      }
```

Comment:
Fetch all users from api rather than from local storage.
try using a map here instead of for loops

Reply:
resolved